src/utils/utils.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable import/prefer-default-export */
+
+// function expects a canvas with variable width, a fixed positive ion radius, and a fixed distance between two positive ions
+// it calculates the number of positive ions that can be placed in one row on the canvas 
+export const calculateNumberOfPositiveIons = (
+  stageWidth,
+  positiveIonRadius,
+  distanceBetweenPositiveIons,
+) => {
+  const positiveIonDiameter = positiveIonRadius * 2;
+
+  // if width of canvas ('stageWidth') is smaller than one positive ion's diameter, then no positive ions can fit
+  if (stageWidth < positiveIonDiameter) {
+    return { numberOfPositiveIons: 0, excessWidth: stageWidth };
+  }
+
+  let numberOfPositiveIons = 1;
+  let horizontalLengthOfPositiveIons =
+    numberOfPositiveIons * positiveIonDiameter +
+    (numberOfPositiveIons - 1) * distanceBetweenPositiveIons;
+
+  while (horizontalLengthOfPositiveIons < stageWidth) {
```

Comment:
I would say that using `break` is not elegant by default even though it is more easy to read aha. A pretty way to avoid them is to have a recursive function `f(numberOfPositionIons, horizontalDistance)` and return at each break point. 
But more generally I get why you have this function, but maybe we can go with something more simpler such as responsive -size ions (but I guess it will need some tweaks for the text). Or display them until it overlap the window border (and it does not really matter if it too much ions or not). Maybe instead of dealing with distance between ions we rather decide how many we display per row and per column. Let me know which way we prefer proceeding!

Reply:
Maybe what you can also use for simplification is something like 
`stageWidth / (ionWidth+ionPadding)` which would give approximatively the number of ions for a line, it will not be exact since it consider blank space but is relatively okay to use I think.

Reply:
thanks @pyphilia interesting thoughts.

- agreed that there are other options, like making the radius responsive: fixing the leftIndent and rightIndent, and then fitting N ions in the remaining space. one reason I avoided this is that it would mean adjusting the size and positioning of the Text element embedded in the charge as well, and I did not feel there are material advantages from doing it this way.

- good idea... I used `Math.floor(stageWidth / ionDiameter + ionPadding)` for another solution. see `LatticeRow.js`. I am partial to this since it seems briefer. there is a minor difference between the two approaches but it doesn't make a difference.

- simultaneously slightly cleaned up `calculateNumberOfPositiveIons` in `utils.js`; not sure how I came up with the phrase `horizontalLength` when I could have said `width` 😶

**let me know which in your opinion is nicer**